tests/test_reddit_subreddit_block.py

```diff
@@ -1,27 +1,71 @@
 from collections import defaultdict
 from nio.common.signal.base import Signal
 from nio.util.support.block_test_case import NIOBlockTestCase
-from ..example_block import Example
+from ..reddit_subreddit_block import SubredditFeed
+from ..http_blocks.rest.rest_block import RESTPolling
+from unittest.mock import patch, MagicMock
 
 
-class TestExample(NIOBlockTestCase):
+class TestSubredditFeed(NIOBlockTestCase):
 
     def setUp(self):
         super().setUp()
         # This will keep a list of signals notified for each output
         self.last_notified = defaultdict(list)
 
-    def signals_notified(self, signals, output_id='default'):
-        self.last_notified[output_id].extend(signals)
+    @patch("requests.post")
+    def test_authenticate(self, mock_post):
+        """ Test that headers are properly set in _authenticate. """
+        blk = SubredditFeed()
+        mock_post.return_value = MagicMock()
```

Comment:
Don't even think we need to patch requests.post at all since we mock the authenticate method